Nuke-WebP-PluginTests/LoadWebPImageTests.swift

```diff
@@ -26,24 +26,64 @@ class LoadWebPImageTests: XCTestCase {
 
     func testsFailLoadWebPImage() {
         let exception = XCTestExpectation(description: "not decode webp image")
-        Nuke.Manager.shared.loadImage(with: self.webpImageURL) {[exception] (result) in
-            XCTAssertNil(result.value)
+        Nuke.ImagePipeline.shared.loadImage(with: self.webpImageURL) { (imageResponse, error) in
+            /// It is an error because it does not correspond to WebP
+            guard let e = error else {
+                XCTFail()
+                return
+            }
+
+            print(e)
             exception.fulfill()
         }
         self.wait(for: [exception], timeout: 1)
     }
     
     func testsLoadWebPImage() {
-        let exception = XCTestExpectation(description: "not decode webp image")
-        NukeWebPPlugin.WebPImage.manager.loadImage(with: self.webpImageURL) { (result) in
-            switch result {
-            case .success:
+        let exception = XCTestExpectation(description: "decode webp image")
+        ImageDecoderRegistry.shared.register { (context) -> ImageDecoding? in
+            return WebPImageDecoder.isWebPFormat(data: context.data) ? WebPImageDecoder() : nil
+        }
+
+        Nuke.ImagePipeline.shared.loadImage(with: self.webpImageURL) { (imageResponse, error) in
+            if let e = error {
+                XCTFail(e.localizedDescription)
                 exception.fulfill()
-            case .failure(let error):
-                XCTFail(error.localizedDescription)
+                return
             }
+            
+            XCTAssertNotNil(imageResponse?.image)
+            exception.fulfill()
         }
+
         self.wait(for: [exception], timeout: 1)
     }
 
+    func testsProgressWebPImage() {
+        let exception = XCTestExpectation(description: "decode progressive webp image")
+        ImageDecoderRegistry.shared.register { (context) -> ImageDecoding? in
+            return WebPImageDecoder.isWebPFormat(data: context.data) ? WebPImageDecoder() : nil
+        }
+
+        let pipeline = ImagePipeline {
+            $0.isProgressiveDecodingEnabled = true
+        }
+        let task = pipeline.loadImage(with: webpImageURL) { (imageResponse, error) in
```

Comment:
I'd suggest to test WebPImageDecoder in isolation instead of testing it via ImagePipeline (see how [progressive JPEG decoding is tested](https://github.com/kean/Nuke/blob/master/Tests/ImageDecoderTests.swift) in a main repo). I've [already written ImagePipeline tests](https://github.com/kean/Nuke/blob/nuke7/Tests/ImagePipelineProgressiveDecodingTests.swift) that verify that progressive decoding works in a main repo.

Reply:
@kean I got it to be a test reference!
Thanks to you I was able to find an element that crashes one safely!
Thanks!